_observing-your-data/alerting/dashboards-alerting.md

```diff
@@ -9,84 +9,297 @@ nav_order: 50
+
+### Alerting query structure
+
+Here is an example of the alerting query structure:
+
+```json
+{
+  "name": "Query Name",
+  "description": "Query Description",
+  "schedule": {
+    "interval": {
+      "period": 10,
+      "unit": "MINUTES"
+    }
+  },
+  "trigger": {
+    "condition": {
+      "script": {
+        "source": "
+          // Trigger condition logic
+        "
+      }
+    }
+  },
+  "input": {
+    "search": {
+      "request": {
+        "indices": ["index-name"],
+        "body": {
+          // Search query
+        }
+      }
+    }
+  },
+  "actions": [
+    {
+      "name": "Action Name",
+      "type": "action-type",
+      "action-type": {
+        // Action configuration
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+In this example:
+
+- The `monitor` runs every 10 minutes (`"period": 10, "unit": "MINUTES"`).
+- The `trigger` object specifies the conditions that, when met, raise an alert. The condition is defined using a script, which can access the search query results through the `ctx.payload` object.
+- The `input` object contains the search query that will be executed by the monitor. The query is defined using [OpenSearch Query DSL]({{site.url}}{{site.baseurl}}/query-dsl/) and can include aggregations, filters, and other query components.
+- The `actions` array specifies the notifications that are sent when the trigger condition is met. Each action has a type (Slack, email, webhook) and a corresponding configuration object. 
+
+#### Query examples
+
+The following example queries show how you can use the alerting query structure to monitor various metrics and conditions in your OpenSearch data. By adjusting the trigger conditions, input queries, and actions, you can create tailored alerting rules to suit your specific monitoring needs.
+
+**High CPU usage alert**
+```json
+{
+  "name": "High CPU Usage Alert",
+  "description": "Alerts when the average CPU usage exceeds 80% in the last hour",
+  "schedule": {
+    "interval": {
+      "period": 1,
+      "unit": "HOURS"
+    }
+  },
+  "trigger": {
+    "condition": {
+      "script": {
+        "source": "
+          double avgCpuUsage = ctx.payload.aggregations.avg_cpu_usage.value;
+          return avgCpuUsage > 80;
+        "
+      }
+    }
+  },
+  "input": {
+    "search": {
+      "request": {
+        "indices": ["system-metrics"],
+        "body": {
+          "size": 0,
+          "aggs": {
+            "avg_cpu_usage": {
+              "avg": {
+                "field": "cpu_usage"
+              }
+            }
+          }
+        }
+      }
+    }
+  },
+  "actions": [
+    {
+      "name": "Notify on Slack",
+      "type": "slack",
+      "slack": {
+        "message": "High CPU usage detected: {{ctx.payload.aggregations.avg_cpu_usage.value}}%"
+      }
+    }
```

Comment:
As mentioned above, the action object isn't structured correctly.

Reply:
@AWSHurneyt Could you provide a working example for this? @vagimeli: Please use the raw/endraw tags when referencing a mustache variable. For example
```
{% raw %}{ "text": "Monitor {{ctx.monitor.name}} just entered alert status. Please investigate the issue. - Trigger: {{ctx.trigger.name}} - Severity: {{ctx.trigger.severity}} - Period start: {{ctx.periodStart}} - Period end: {{ctx.periodEnd}}" }{% endraw %}
```

Reply:
This example has an action with a `message_template` that references mustache variables.
https://opensearch.org/docs/latest/observing-your-data/alerting/api/#bucket-level-monitors